src/components/CreateCourse/tests/createCourse.test.js

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+
+// eslint-disable-next-line import/order
+import { render, fireEvent } from '@testing-library/react';
+// eslint-disable-next-line import/order
+import { createMemoryHistory } from 'history';
+// eslint-disable-next-line import/order
+import { Router } from 'react-router';
+import * as actions from '../../../store/authors/actionsCreators';
+
+import '@testing-library/jest-dom';
+// eslint-disable-next-line import/order
+import { Provider } from 'react-redux';
+
+import CreateCourse from '../CreateCourse';
+// eslint-disable-next-line import/order
+import initialState from '../../../mocks/initilState';
+
+const buildComponent = (store) => {
+	const history = createMemoryHistory();
+
+	return render(
+		<Router history={history}>
+			<Provider store={store}>
+				<CreateCourse />
+			</Provider>
+		</Router>
+	);
+};
+
+jest.mock('react-router-dom', () => ({
+	...jest.requireActual('react-router-dom'),
+
+	useParams: () => ({
+		id: 'course-id-1',
+	}),
+}));
+
+describe('testing CreateCourse component', () => {
+	const mockedStore = {
+		getState: () => initialState,
+		dispatch: jest.fn(),
+		subscribe: jest.fn(),
+	};
+
+	test('Should show authors lists', () => {
+		const { container } = buildComponent(mockedStore);
```

Comment:
Move it inside `describe` no need to write the same code in every test case
```suggestion
		const { container } = buildComponent(mockedStore);
```

Reply:
Yes, it's a great idea, but it will not work.
Because, for the each test we have to render itself DOM container in scope of test, but I am getting your drift.

Reply:
As I can see from current branch version, it definitely possible to avoid duplication here. ;)